### PR TITLE
Hide join call button when the user is already in the call

### PR DESCRIPTION
**Pull request by bmarty**
## Content

When the user is in the call due to the Pip mode, the "Join" call button is displayed. This PR hides it if the user is currently (and locally) in the call.

This PR also improve the `RoomCallState` regarding permission. We now have a disabled button if the user is not allowed to join the call.

## Motivation and context

Closes https://github.com/element-hq/element-meta/issues/2608

## Screenshots / GIFs

See the recorded ones.

## Tests

- In a room, start a call
- Enter Pip mode
- Observe that the button to join the call in the room (top end of the screen) is not displayed, and it is also not displayed anymore in the timeline item of the call.

Note that if there is a call in another room, joining the other call will do nothing. This is another issue, not related to this PR.

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/call/api/src/main/kotlin/io/element/android/features/call/api/CurrentCallObserver.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Please see LICENSE in the repository root for full details.
+ */
+
+package io.element.android.features.call.api
+
+import kotlinx.coroutines.flow.StateFlow
+
+interface CurrentCallObserver {
```
Comment (ganfra) [inline features/call/api/src/main/kotlin/io/element/android/features/call/api/CurrentCallObserver.kt:12]: The name is a bit weird, the observer is the one who subscribe to the flow usually, not the one providing.
I'd call this CurrentCallService or CurrentCallDataSource for exemple

File: features/call/test/src/main/kotlin/io/element/android/features/call/test/FakeCurrentCallObserver.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+package io.element.android.features.call.test
+
+import io.element.android.features.call.api.CurrentCall
+import io.element.android.features.call.api.CurrentCallObserver
+import kotlinx.coroutines.flow.MutableStateFlow
+
+class FakeCurrentCallObserver(
+    initialValue: CurrentCall = CurrentCall.None,
+) : CurrentCallObserver {
+    override val currentCall = MutableStateFlow(initialValue)
```
Comment (ganfra) [inline features/call/test/src/main/kotlin/io/element/android/features/call/test/FakeCurrentCallObserver.kt:17]: Usually when we have `StateFlow` we provide them in the constructor of the fake

**Review by ganfra — APPROVED** [review]
Small remarks otherwise LGTM

File: features/call/api/src/main/kotlin/io/element/android/features/call/api/CurrentCallObserver.kt (hunk 1 above)
Comment (bmarty) [reply features/call/api/src/main/kotlin/io/element/android/features/call/api/CurrentCallObserver.kt:12]: 2b5acb30232d026fc71f9e7e00a3c228ce370cb3

File: features/call/test/src/main/kotlin/io/element/android/features/call/test/FakeCurrentCallObserver.kt (hunk 2 above)
Comment (bmarty) [reply features/call/test/src/main/kotlin/io/element/android/features/call/test/FakeCurrentCallObserver.kt]: db4b4d3fa4e3434e13a2c7af4ee7abb578304c5f

**Pull request merged**